### ISSUE TITLE
Build not triggered when merging

### ISSUE THREAD

**Issue by milhomem** [cirruslabs/cirrus-ci-docs#487]
# Question

How to figure it out why my build is not triggering?

I use Git**Lab** and I was forced to set up a repository mirroring to Git**Hub**.

My .cirrus.uml contains this condition to only process the `master` branch
```yaml
task:
  only_if: $CIRRUS_BRANCH == 'master'
```

It works fine if I push manually to master on Git**Lab**, but if I merge an MR with a fast-forward strategy it does not trigger. Although I can see the the new commits in Git**Hub**

**Comment by fkorotkov** [comment]
@milhomem is it for a public repository? In any case what is the full name of the repository? It might help to debug the issue.

**Comment by milhomem** [comment]
Hi @fkorotkov , no, it's not public.

Repository url: https://github.com/ManageFlitter/ukufu-np-mobile-app.

**Comment by fkorotkov** [comment]
I see some errors that indicate that Cirrus wasn't able to find `.cirrus.yml` for some of the SHAs. 🤔 Do you mind to share a master SHA that a build wasn't triggered for?

**Comment by milhomem** [comment]
Sure, SHAs that didn't work:
```
0ba81c4 
fcc2d11f
75bc842 (When .cirrus.yml was introduced)
```

SHAs that worked with manual push:
```
670e6977
```

**Comment by fkorotkov** [comment]
Haven't found anything suspicious. Don't see these SHAs in Cirrus logs anywhere. How did you configure the sync? Like in [GitLab docs](https://docs.gitlab.com/ee/workflow/repository_mirroring.html#setting-up-a-push-mirror-from-gitlab-to-github-core)? I'll try to reproduce on a test repository.

**Comment by milhomem** [comment]
Hi @fkorotkov , thanks for checking.

Yes, I configured the repository to do a Push mirroring sync as described in [here](https://docs.gitlab.com/ee/workflow/repository_mirroring.html#pushing-to-a-remote-repository-core) with an SSH [deploy key](https://developer.github.com/v3/guides/managing-deploy-keys/#deploy-keys) with write permission copied from [here](https://docs.gitlab.com/ee/workflow/repository_mirroring.html#ssh-public-key-authentication).

**Comment by fkorotkov** [comment]
Hm. I've tried to reproduce but it worked for me. Seems I've followed your instructions and has added a generated by Git**Lab** SSH key as a deploy key to Git**Hub**:

![image](https://user-images.githubusercontent.com/989066/66791984-8bebd500-eec4-11e9-868e-b7876ec5cc96.png)

And here are build: https://cirrus-ci.com/github/fkorotkov/gitlab-playground

**Comment by fkorotkov** [comment]
I'll try with private repositories now. Maybe that will make a difference. 🤷‍♂

**Comment by fkorotkov** [comment]
Worked on a private personal. 🤔

**Comment by milhomem** [comment]
Hi @fkorotkov  are you merging a PR? 

Our merge strategy is `Fast-forward merge`. It works when I push directly to the branch but does not work when I open a PR and merge it to master.

**Comment by milhomem** [comment]
I also noticed your `.cirrus.yaml` does not contain:

`task:
  only_if: $CIRRUS_BRANCH == 'master'`

Which prevents the build to happen when you push the branch, but should allow when pushing to master and ideally when merging an MR.

**Comment by fkorotkov** [comment]
Hm. Tried an MR now and also worked.  :-( Will take a look tomorrow on a fresh head. Just to confirm: you do see the commits mirrored on GitHub? And you don't see any errors from GitLab side?

Another quick question: do you use the same email both on GitHub and GitLab? Do you see yourself as the author of mirrored commits?

**Comment by milhomem** [comment]
> you do see the commits mirrored on GitHub?

Yes, the new commits were there on master after the MR was merged.

> And you don't see any errors from GitLab side?

No, GitLab report as synced.

> do you use the same email both on GitHub and GitLab?

I use the company email as a committer, which is different from my GitHub profile. My GitLab profile email is different from the committer and from GitHub too.

> Do you see yourself as the author of mirrored commits?

Yes, I see my email company as author in GitHub with the same git log tree as in GitLab.

**Comment by milhomem** [comment]
@ChubV can you please share the lastest merges and a sanitized `git log` output of `master`?

**Comment by ChubV** [comment]
```
f42da01 (HEAD -> master, origin/master, origin/improve-http, origin/HEAD) Mov...
cd273e4 Ref....
ff46f7c Imp....
4a25730 Mov....
91255e0 Imp....
e90c68b Han....
45c0785 Imp....
5491ec3 Imp....
2bd7d44 Imp....
0ba81c4 Ign....
fcc2d11 Fix....
670e697 Tes....
75bc842 Int....
```

**Comment by fkorotkov** [comment]
Don't see any of the SHAs in the logs. I do see though that Cirrus is getting webhook events for some SHAs of `ManageFlitter/ukufu-np-mobile-app`. For example:

```
1baa5cedfd9db66f4c813905306687616c1c89a6
1c91ee471bc5fe96b255cb80326664bbaa3e42a9
5a0b02904421616ab4bff43008811e7f3f75c145
```

But seems these SHAs are for non-master branch and no builds are created. So seems that Cirrus is getting events for all branches but `master`. 🤔 I wonder what it could be... Could you please confirm that by guess is correct and that the above SHAs are for non-master branches?

**Comment by fkorotkov** [comment]
In the meantime since Cirrus doesn't store events from GitHub and GitHub doesn't have tools to search through the events. I've added some logs that might help with the investigation.

**Comment by ChubV** [comment]
these 3 SHAs from sub-branches, not master

**Comment by ChubV** [comment]
`dae2027d` was recently pushed to subbranch first and then to master (fast-forward).

**Comment by fkorotkov** [comment]
Seems it's triggering now. Sorry for the inconvenience and thanks for helping to debug the corner case.
